src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Chain.cs

```diff
@@ -112,6 +112,20 @@ internal bool Build(X509Certificate2 certificate, bool throwOnException)
                 if (certificate == null || certificate.Pal == null)
                     throw new ArgumentException(SR.Cryptography_InvalidContextHandle, nameof(certificate));
 
+                if (_chainPolicy != null && _chainPolicy.CustomTrustStore != null)
+                {
+                    if (_chainPolicy.TrustMode == X509ChainTrustMode.System && _chainPolicy.CustomTrustStore.Count > 0)
+                        throw new CryptographicException(SR.Cryptography_CustomTrustCertsInSystemMode, nameof(_chainPolicy.TrustMode));
+
+                    foreach (X509Certificate2 customCertificate in _chainPolicy.CustomTrustStore)
+                    {
+                        if (customCertificate == null || customCertificate.Handle == IntPtr.Zero)
+                        {
+                            throw new ArgumentException(SR.Cryptography_InvalidTrustCertificate, nameof(_chainPolicy.CustomTrustStore));
```

Comment:
This one should also be a CryptographicException, since it's not about an argument to this method.